### PR TITLE
add a gcov target

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -107,7 +112,11 @@ pytest-valgrind: all
 
 endif
 
+ifdef WITH_GCOV
+test: all pytest
```
Comment (swapnili) [inline Makefile:116]: nit: tab

Comment (jlevon) [reply Makefile:116]: this is a rule, I don't think we want it indented

File: lib/CMakeLists.txt [hunk 2]
```diff
@@ -65,24 +65,13 @@ set_target_properties(vfio-user-static PROPERTIES
...
-
-add_library_ut(pci_caps pci_caps.c)
-add_library_ut(dma dma.c)
-add_library_ut(irq irq.c)
-add_library_ut(libvfio-user libvfio-user.c)
-add_library_ut(migration migration.c)
-add_library_ut(pci pci.c)
-add_library_ut(tran_sock tran_sock.c)
+add_library(pci_caps OBJECT pci_caps.c)
+add_library(dma OBJECT OBJECT dma.c)
```
Comment (swapnili) [inline lib/CMakeLists.txt:69]: Double OBJECT type?

File: lib/CMakeLists.txt [hunk 3]
```diff
@@ -65,24 +65,13 @@ set_target_properties(vfio-user-static PROPERTIES
...
-endfunction(add_library_ut)
-
-add_library_ut(pci_caps pci_caps.c)
-add_library_ut(dma dma.c)
-add_library_ut(irq irq.c)
-add_library_ut(libvfio-user libvfio-user.c)
-add_library_ut(migration migration.c)
-add_library_ut(pci pci.c)
-add_library_ut(tran_sock tran_sock.c)
+add_library(pci_caps OBJECT pci_caps.c)
```
Comment (swapnili) [inline lib/CMakeLists.txt:68]: Can all of these add_library() moved closer to set(LIBOBJS, ) at line:40 ?
This way if someone adds new file then its clear to also update set(LIBOBJS)

File: Makefile (hunk 1 above)
Comment (swapnili) [reply Makefile:116]: Aah yes

**Review by swapnili — APPROVED** [review]

**Pull request merged**